plugins/scss/index.js

```diff
@@ -173,24 +186,46 @@ const writer = function(thisPackage, outputDir) {
     console.log('Writing ' + file);
     promises.push(fs.writeFileAsync(file, args.join(' ')));
 
+    var themeDir = outputDir + '/themes';
+    var fontsDir = themeDir + '/fonts';
+    if (fonts && fonts.length) {
+      promises.push(mkdirp.mkdirpAsync(fontsDir)
+        .then(() => {
+          var fontPromises = [];
+          fonts.forEach(function(font) {
+            var fontFolder = path.resolve(fontsDir, font);
+            fontPromises.push(mkdirp.mkdirpAsync(fontFolder)
+              .then(() => {
+                var fontPath = utils.resolveModulePath(font);
+                console.log('Copying ' + fontPath + ' to ' + fontFolder);
+                var writePromises = [];
+                writePromises.push(fse.copy(fontPath + '/files', fontFolder + '/files'));
+                writePromises.push(fs.copyFileAsync(fontPath + '/index.css', fontFolder + '/index.css'));
+                return Promise.all(writePromises);
+              }));
+          });
+          return Promise.all(fontPromises);
+        }));
+    }
+
```

Comment:
Why is this copying font resources here instead of having projects using the `resources` plugin via `index.js` which already exists for that purpose?

Reply:
That's how it is today with opensphere / other applications. However that isn't practical when you have dependency repos providing fonts, as the applications will have to have those fonts in their index.js. Even more of a problem when the dependency repo changes the fonts it requires. That would defeat the purpose of a specific dependency repo providing fonts. This locates all fonts / themes into the build directory, where then the application index.js / package.json can copy to the dist directory


see https://github.com/ngageoint/opensphere/pull/969 as an example

Reply:
I attempted to re-write this solution using the resources plugin. Due to how fontloader.scss works in opensphere, it wont be able to grab fonts from npm OR yarn node_modules. So we have to include the fonts in the repo (as we do today). Also dependencies cannot provide themes with the resolver today. This PR is both allowing dependencies to provide themes and also fonts. It makes sense to keep fonts with the scss rather than splitting it into the normal js dependencies because the scss includes the fonts via an import. Normal resources arent referenced in the scss, and just added to the index.html files.

I still stand by this PR, I think its a solid approach to the problem